u2f-ref-code/java/src/com/google/u2f/server/impl/attestation/android/AndroidKeyStoreAttestation.java

```diff
@@ -49,7 +49,10 @@ private AndroidKeyStoreAttestation(Integer keymasterVersion, byte[] attestationC
   }
 
   /**
-   * Parses the key description extension.  Expected format is:
+   * Parses the key description extension.  Note that this method only parses the description
+   * extension in the leaf cert.  It *does not* validate the certificate (or any chain).
```

Comment:
About validating the cert chain: Yes, somewhere. This comment seems to indicate a warning to its callers. If you expect this method will eventually begin validating the whole cert chain, a comment here would be nice, so the callers who read these comments know this is a temporary state. If you intend to add a separate method, I'd add a comment wherever you expect it to land.

About the UI elements: no, I think that doesn't require a TODO as much. The rough metric I'm applying is, 1) the code base should be consistent at all points in time, and 2) the comments should guide the reader, especially warning about potential gotchas. The UI being not feature complete isn't a consistency issue, and doesn't pose any particular gotcha, that's why I don't think it needs one.